sklearn/cluster/_kmeans.py

```diff
@@ -1173,9 +1174,9 @@ class KMeans(_BaseKMeans):
 
         'k-means++' : selects initial cluster centroids using sampling based on
         an empirical probability distribution of the points' contribution to the
-        overall inertia. This technique speeds up convergence, and is
-        theoretically proven to be :math:`\\mathcal{O}(\\log k)`-optimal.
-        See the description of `n_init` for more details.
+        overall inertia. This technique speeds up convergence. The algorithm 
+        implemented is "greedy k-means++". It differs from the original k-means++
+        by making several trials at each sampling step and chosing the best centroid among them.
```

Comment:
Let's use "vanilla" instead of "original" because both variants were introduced in the original k-means++ paper.

```suggestion
        overall inertia. This technique speeds up convergence. The algorithm 
        implemented is "greedy k-means++". It differs from the vanilla k-means++
        by making several trials at each sampling step and choosing the best centroid
        among them.
```